### ISSUE TITLE
Equality of polynomial types seems broken

### ISSUE THREAD

**Issue by jishnub** [JuliaMath/Polynomials.jl#408]
Currently, equality of polynomial types checks their coefficients. Ideally, they should be converted to `Polynomial` or `LaurentPolynomial` types before their coefficients are compared. Otherwise, we end up with
```julia
julia> p1 = ChebyshevT([1,2,3])
ChebyshevT(1⋅T_0(x) + 2⋅T_1(x) + 3⋅T_2(x))

julia> p2 = Polynomial([1,2,3])
Polynomial(1 + 2*x + 3*x^2)

julia> p1 == p2 # shouldn't hold
true

julia> p1(0.5) == p2(0.5)
false
```

The current method:
https://github.com/JuliaMath/Polynomials.jl/blob/8b1c45a3403013095f88f0b7ac68f54afea8bf5c/src/common.jl#L1021-L1022

this should only hold for `StandardBasisPolynomial`s I think

**Comment by jverzani** [comment]
Thanks! Indeed that isn't general enough. I wonder what is a better design. My first thought says the best design would be to promote p1 and p2 to a common type and then compare as is done (though there would need to be special cases for LaurentPolynomials and SparsePolynomials.

**Comment by jishnub** [comment]
I agree that promoting to a common type seems to be a good approach. I guess there might be some subtlety involved in floating-point precision loss

**Comment by jverzani** [comment]
Yes, always the case. I'd suggest `\approx` then, but when looking that over, the conversion is subtle, as it happens in calling norm. Need to fix that as well.

**Comment by jverzani** [comment]
closed by #410
